src/compiler/zx_calculus.jl

```diff
@@ -1,14 +1,72 @@
-using ZXCalculus
+using ZXCalculus, LightGraphs
+using ZXCalculus: qubit_loc
 
 function optimize!(ir::YaoIR)
     circ = to_ZX_diagram(ir)
-    clifford_simplify!(circ)
-    ir = to_YaoIR(circ)
+    circ = clifford_simplify(circ)
+    ir.body = to_YaoIR(circ)
     return ir
 end
 
+function to_YaoIR(circ::ZXDiagram{T, P}) where {T, P}
+    lo = circ.layout
+    vs = spiders(circ)
+    locs = Dict()
+    nqubit = lo.nbits
+    frontier_v = ones(T, nqubit)
+    ex = quote
+        $(Expr(:meta, :register, :new, gensym(:register)))
+        return
+    end
+    lowered_ast = Meta.lower(@__MODULE__, ex)
+    ir = mark_quantum(IRTools.IR(lowered_ast.args[], 0))
```

Comment:
this could segment fault, if you don't update slots.

Reply:
mark quantum is not necessary here if you construct the `YaoIR` object body directly from an empty `IR` object

Reply:
Empty `IR` has no register and return. So I use these codes for generating them.

Reply:
you can `push!` these two argument as a statement and branch to `IR`. This will cause a lot extra work and make the compilation slow.